card/lib/card/format/error.rb

```diff
@@ -1,6 +1,40 @@
 class Card
   class Format
     module Error
+      def ok? task
```

Comment:
We don't have format based permission rules. Why make it a format method? 

In my pending PR I have to check permissions for another card. With this change I have to create a format object for that (https://github.com/decko-commons/decko/pull/360/files#diff-2ed9b8393ffd5a7e60ce5dc1ec38ddeeR63)

Reply:
This is actually not a change.  It's just moved here from format/permission.rb (the rest of which was moved into card/view/permission.rb).

The real permission handling is still on cards.  This method's main job is caching.  This way you don't have to look up the same permission many times for the same format.  

The create/update logic is a little nuanced here (again, not new, but we can revisit it and should probably comment on it).  It's saying that if a view requires update permission but the card is new, then you should have to have create permission to see it.  

In principle we could do this local caching on the card instead of the format, but we'd need to make sure we cached the as_user, too.  (I don't think we ever use "as_bot" in view rendering; we use the `_render` (or `skip_perms`) pattern instead, so we don't have to have it here).